### PR TITLE
Add Camera fisheye effect

### PR DESCRIPTION
**Pull request by stefaniapedrazzi**
Continues from #5240.

Tasks:
* [x] Improve `merge_spherical.frag` shader to produce fisheye effect
* [x] Replace `Camera.spherical` field with `Camera.projection` to enable both spherical and cylindrical projection
* [x] Remove old deprecated Camera fields
* [x] Update documentation
* [x] Add entry in Changelog
* [x] Update tests
* [x] Add new test

### CODE REVIEW

File: src/webots/nodes/WbCamera.cpp [hunk 1]
```diff
@@ -107,16 +107,12 @@ void WbCamera::init() {
...
-    parsingWarn("Deprecated 'colorNoise' field, please use the 'noise' field instead.");
-    if (mNoise->value() == 0.0)
-      mNoise->setValue(colorNoise->value());
-    colorNoise->setValue(0.0);
+  WbSFBool *sphericalField = findSFBool("spherical");
+  if (sphericalField->value()) {  // Introduced in Webots R2023
+    parsingWarn("Deprecated 'spherical' field, please use the 'projection' field instead.");
+    if (isPlanarProjection())
+      mProjection->setValue("cylindrical");
+    sphericalField->setValue(false);
```
Comment (omichel) [inline src/webots/nodes/WbCamera.cpp:115]: Why do you set it to false here? If the user saves, it will be FALSE while they didn't changed anything...

File: src/webots/nodes/WbCamera.cpp [hunk 2]
```diff
@@ -107,16 +107,12 @@ void WbCamera::init() {
...
-    parsingWarn("Range finder type is not available for camera since Webots 8.4, please use a RangeFinder node instead.");
-
-  WbSFDouble *colorNoise = findSFDouble("colorNoise");
-  if (colorNoise->value() != 0.0) {  // Introduced in Webots 8.4.0
-    parsingWarn("Deprecated 'colorNoise' field, please use the 'noise' field instead.");
-    if (mNoise->value() == 0.0)
-      mNoise->setValue(colorNoise->value());
-    colorNoise->setValue(0.0);
+  WbSFBool *sphericalField = findSFBool("spherical");
+  if (sphericalField->value()) {  // Introduced in Webots R2023
```
Comment (omichel) [inline src/webots/nodes/WbCamera.cpp:111]:
```suggestion
  if (sphericalField->value()) {  // Deprecated in Webots R2023
```

File: src/webots/nodes/WbLidar.cpp [hunk 3]
```diff
@@ -81,6 +81,15 @@ void WbLidar::init() {
 
   mTcpImage = NULL;
   mTcpCloudPoints = NULL;
+
+  // backward compatibility
+  WbSFBool *sphericalField = findSFBool("spherical");
+  if (!sphericalField->value()) {  // Introduced in Webots R2023
```
Comment (omichel) [inline src/webots/nodes/WbLidar.cpp:87]:
```suggestion
  if (!sphericalField->value()) {  // Deprecated in Webots R2023
```

File: src/webots/nodes/WbLidar.cpp [hunk 4]
```diff
@@ -81,6 +81,15 @@ void WbLidar::init() {
...
+    parsingWarn("Deprecated 'spherical' field, please use the 'projection' field instead.");
+    if (mProjection->value() == "cylindrical")
+      mProjection->setValue("planar");
+    sphericalField->setValue(true);
```
Comment (omichel) [inline src/webots/nodes/WbLidar.cpp:91]: Why this change?

File: src/webots/nodes/WbRangeFinder.cpp [hunk 5]
```diff
@@ -33,6 +33,15 @@ void WbRangeFinder::init() {
   mMinRange = findSFDouble("minRange");
   mMaxRange = findSFDouble("maxRange");
   mResolution = findSFDouble("resolution");
+
+  // backward compatibility
+  WbSFBool *sphericalField = findSFBool("spherical");
+  if (sphericalField->value()) {  // Introduced in Webots R2023
```
Comment (omichel) [inline src/webots/nodes/WbRangeFinder.cpp:39]:
```suggestion
  if (sphericalField->value()) {  // Deprecated in Webots R2023
```

File: src/webots/nodes/WbRangeFinder.cpp [hunk 6]
```diff
@@ -33,6 +33,15 @@ void WbRangeFinder::init() {
...
+    parsingWarn("Deprecated 'spherical' field, please use the 'projection' field instead.");
+    if (isPlanarProjection())
+      mProjection->setValue("cylindrical");
+    sphericalField->setValue(false);
```
Comment (omichel) [inline src/webots/nodes/WbRangeFinder.cpp:43]: Why this change?

**Review by omichel — CHANGES_REQUESTED** [review]
Generally, it looks very good to me.
I only have a few questions/requests.
Also, I understand it makes no sense to have a spherical projection for a lidar, but for the range-finder, this is more arguable. What is the motivation for making the range-finder only planar or cylindrical and not spherical?

File: docs/reference/rangefinder.md [hunk 7]
```diff
@@ -49,9 +49,10 @@ Since range-finder pixels are squares, the vertical field of view can be compute
 
 - `height`: height of the image in pixels
 
-- `spherical`: switch between a planar or a spherical projection.
-A spherical projection can be used for example to simulate a lidar device.
-More information on spherical projections is provided in the [spherical projection](camera.md#spherical-projection) section of the [Camera](camera.md) node.
+- `projection`: switch between a planar or a cylindrical projection.
+A cylindrical projection can be used for example to simulate a lidar device.
```
[the remark this answers is not in the document]
Comment (stefaniapedrazzi) [reply docs/reference/rangefinder.md:53]: Yes. I agree.
This was already in the documentation and I didn't add it.

Note that there are at least another reference to the lidar in the RangeFinder documentation.
This should be fixed in the released branch.

File: src/webots/nodes/WbCamera.cpp (hunk 1 above)
Comment (stefaniapedrazzi) [reply src/webots/nodes/WbCamera.cpp:115]: This is the backward compatibility method.
When loading the node we want to convert the settings of the deprecated `spherical` field to the new `projection` field.
So we exactly want to reset the `spherical` field because should no longer be used.
Even if the user doesn't change nothing, the saved world should be now compatible with R2023c and the `projection` field should be set and the `spherical` field should be reset to default.

**Comment by stefaniapedrazzi** [comment]
> What is the motivation for making the range-finder only planar or cylindrical and not spherical?

I think the shader was only tested with RGB camera.
I will check it for range-finder images too.

File: src/webots/nodes/WbCamera.cpp (hunk 1 above)
Comment (omichel) [reply src/webots/nodes/WbCamera.cpp:115]: OK, that makes sense. Thank you for the explanation.

**Comment by stefaniapedrazzi** [comment]
I enabled the spherical projection for the range-finder camera and fixed preventing using it for lidars.

Windows compilation is failing due to errors with OpenAL but they are not related to this PR (same errors in the scheduled nightly tests on master).

File: docs/reference/rangefinder.md [hunk 8]
```diff
@@ -49,9 +49,8 @@ Since range-finder pixels are squares, the vertical field of view can be compute
 
 - `height`: height of the image in pixels
 
-- `spherical`: switch between a planar or a spherical projection.
-A spherical projection can be used for example to simulate a lidar device.
-More information on spherical projections is provided in the [spherical projection](camera.md#spherical-projection) section of the [Camera](camera.md) node.
+- `projection`: switch between a planar or a cylindrical projection.
```
Comment (omichel) [inline docs/reference/rangefinder.md:52]:
```suggestion
- `projection`: switch between a planar, a cylindrical or a spherical projection.
```

**Review by omichel — APPROVED** [review]
Thank you.

**Comment by omichel** [comment]
It's nice to see that the Windows build appear to be repaired 😄.
